signer/src/storage/postgres.rs

```diff
@@ -1477,34 +1477,21 @@ impl super::DbRead for PgStore {
 
     async fn get_pending_accepted_withdrawal_requests(
```

Comment:
Shouldn't this also filter out expired requests?

Reply:
I'm not sure about what that could return, nor about the possible strange configurations that we may end up with, nor about off by one errors, and that's kinda why I was thinking doing an explicit check may put us in a better situation. It also ensure that this is aligned with the other checks we do related to expiration around (eg in other queries, or in bitcoin validation).

Reply:
Are you referring to this check from your rejection query?
```sql
                JOIN sbtc_signer.bitcoin_blocks block
                  ON block.block_hash = parent.bitcoin_anchor
                WHERE block.block_height >= (SELECT MIN(block_height) FROM bitcoin_blockchain)
```

Reply:
Yeah, as well as the check in `WithdrawalRequestReport` and `RejectWithdrawalV1` that are based on the height diff instead of the anchored block.

Reply:
Good catch, yeah we need to change this.

Reply:
Alright, made some changes in [fdcd265](https://github.com/stacks-network/sbtc/pull/1397/commits/fdcd2654b5f51e69920377f30203e7ae9f391ba4).

The approach here is to:
- Retrieve all withdrawal requests, regardless of chain, whose bitcoin height is >= the provided min height.
- Fetch the canonical bitcoin chain until the lowest height of the retrieved withdrawal requests
- Fetch the canonical stacks chain until the same anchor height
- The rest is pretty much the same, just grouped some of the joins together with the withdrawal requests CTE.

Note that this is used together with `WITHDRAWAL_BLOCKS_EXPIRY`; the `WITHDRAWAL_EXPIRY_BUFFER` check is done in code (where we log that it's on its way to expiring) -- that's in the other PR.

Reply:
This now handles `N-1` per the Slack answers we got: [d49a0c6](https://github.com/stacks-network/sbtc/pull/1397/commits/d49a0c603f42980fb1745e3d198bd9f1851b9a52)